### DISCUSSION TITLE
component issue

### DISCUSSION THREAD

**Discussion by satishjamle** [Mayank-Garg7/typescript_practice#5] [Q&A] [answered]
import React from 'react'

let currDate = new Date();
currDate = currDate.getHours();

export const Date = () => {
  return (
    <>
    <h1> Hello sir , {currDate}</h1>
    
    
    </>
  )
}

in above component i am facing problem could you please tell me why  and how to resolve this error

**Comment by Mayank-Garg7** [comment, marked as answer]
I think there is an issue with the component name, so I'm attaching a list of keywords below to prevent this kind of error from happening again. So check that list below 

---

# Avoiding Reserved Keywords for React Component Names

It seems the issue with your component named `Date` may not be due to it being a reserved keyword but rather a conflict with the built-in `Date` object in JavaScript, which is a global object. While `Date` is not technically a reserved keyword, it’s strongly recommended to avoid using it as a component name to prevent confusion or unintended behavior.

Below is a guide to help you identify and avoid reserved or commonly used names that could cause issues when naming React components.

---

## React-Reserved Keywords to Avoid

React has certain reserved keywords and commonly used names integral to its functionality. Using these names for components can lead to errors or unexpected behavior. Here’s a list of React-reserved keywords to avoid:

### **Core React API Components and Hooks**
- `React`
- `Component`
- `useState`
- `useEffect`
- `useContext`
- `useReducer`
- `useRef`
- `useMemo`
- `useCallback`

### **React-Specific Features**
- `Fragment`
- `StrictMode`
- `Suspense`
- `createContext`
- `createRef`
- `forwardRef`
- `lazy`
- `memo`
- `hydrate`
- `render`
- `setState`

### **React Lifecycle Methods**
- `getDerivedStateFromProps`
- `componentDidMount`
- `componentDidUpdate`
- `componentWillUnmount`
- `shouldComponentUpdate`

---

## Why Avoid Certain Names?

1. **Conflicts with Built-in JavaScript Objects**  
   Avoid using names like `Date`, `Promise`, or `Map`, as they are part of the JavaScript global object and can lead to unexpected results or confusion.

2. **Confusion with HTML Elements**  
   Using names like `div`, `span`, or `button` for components can cause readability issues and confusion between custom components and standard HTML elements.

3. **Potential Overwriting of Core Functions**  
   Reusing React methods (e.g., `setState`) or lifecycle hooks as component names can lead to errors and hinder debugging.

---

## Best Practices for Naming Components

- **Use Descriptive Names**  
  Opt for names that clearly convey the purpose of the component. For example, instead of `Date`, consider using `CustomDatePicker` or `EventDate`.

- **Follow PascalCase Convention**  
  React components should be named in PascalCase (e.g., `MyComponent`), which differentiates them from regular JavaScript functions and HTML tags.


---
